File: backend/gateway/internal/customtype/dto/prescription.go
package dto

import (
	"time"
)

type PrescriptionDTO struct {
	Medication *string    `json:"medication"`
	Dosage     *string    `json:"dosage"`
	Notes      *string    `json:"notes"`
	Started    *time.Time `json:"started" gorm:"type:timestamp;"`
	Ended      *time.Time `json:"ended" gorm:"type:timestamp;"`
	Refills    *int       `json:"refills"`
	Owner      *string    `json:"owner"`
}

File: frontend/guardian/src/app/core/models/rx/Prescription.ts
interface IObjectKeys {
  [key: string]: string | number | null;
}

export interface Prescription extends IObjectKeys {
  id: string;
  medication: string;
  dosage: string;
  notes: string;
  started: string | null;
  ended: string | null;
  refills: string | null;
  owner: string;
}

export const generatePrescriptionTemplate = () => {
  return {
    id: '',
    medication: '',
    dosage: '',
    notes: '',
    started: '',
    ended: '',
    refills: '',
    owner: '',
  };
};
